### PR TITLE
feat: add oneDNN CPU build

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by amontoison** [comment]
@avik-pal Intel doesn't already provide a precompiled version of oneDNN like oneAPI?

**Comment by avik-pal** [comment]
Binaries seem to present via anaconda https://anaconda.org/conda-forge/onednn.

Ones on the intel oneAPI website (https://www.intel.com/content/www/us/en/developer/tools/oneapi/onednn-download.html?operatingsystem=linux&distribution-linux=offline) link to a script

**Comment by avik-pal** [comment]
windows build is failing with "[19:00:39] /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/9.1.0/../../../../x86_64-w64-mingw32/bin/ld: CMakeFiles/dnnl.dir/objects.a(ittnotify_static.c.obj):ittnotify_static.c:(.text+0x527d): undefined reference to `strnlen_s'"

Not quite sure how to fix that

**Comment by giordano** [comment]
I haven't read through all the discussion, but https://sourceforge.net/p/mingw-w64/mailman/mingw-w64-public/thread/AANLkTi=-pNtNaAi=x=GPQMenOEM_ch0zWPpfhH0amLoO@mail.gmail.com/ looks relevant

**Comment by avik-pal** [comment]
all builds seem to pass now.

**Comment by avik-pal** [comment]
Let me know if any changes are needed here.

There is also an older attempt at this https://github.com/JuliaPackaging/Yggdrasil/pull/4550, which seems to have gone stale.

File: O/oneDNN/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+cmake .. ${CMAKE_FLAGS[@]}
+
+if [[ "${target}" == *-mingw* ]]; then
+    # https://oneapi-src.github.io/oneDNN/dev_guide_build.html#windows
+    cmake --build . --config=Release --target install
+else
+    make -j${nproc}
+    make install
+fi
```
Comment (giordano) [inline O/oneDNN/build_tarballs.jl:48]: Why different commands?  Can't just use (https://docs.binarybuilder.org/stable/build_tips/#CMake-builds)
```
cmake --build --parallel ${nproc}
cmake --install
```
everywhere?

**Pull request merged**